### PR TITLE
Fix parallel sql leak

### PR DESCRIPTION
**Pull request by dorinhogea**
bound parameter are repackaged for each parallel thread, and that was leaked in a few scenarios.

### CODE REVIEW

File: db/dohast.c [hunk 1]
```diff
@@ -337,6 +342,13 @@ static void node_free(dohsql_node_t **pnode, sqlite3 *db)
         node_free(&node->nodes[i], db);
     }
 
+    /* params */
+    if ((*pnode)->params) {
+        if ((*pnode)->params->params)
+            free((*pnode)->params->params);
```
Comment (riverszhang89) [inline db/dohast.c]: You don't need to check for NULL. `free(NULL)` is a no-op.

File: db/dohsql.c [hunk 2]
```diff
@@ -1087,6 +1087,8 @@ static void _shard_disconnect(dohsql_connector_t *conn)
     if (conn->cols)
         free(conn->cols);
 
+    if (conn->params)
+        free(conn->params);
```
Comment (riverszhang89) [inline db/dohsql.c]: ditto.

File: db/dohsql.c [hunk 3]
```diff
@@ -1879,8 +1881,10 @@ struct params_info *dohsql_params_append(struct params_info **pparams,
     newparam = clnt_find_param(params->clnt, name + 1, index);
     if (!newparam) {
         /* clnt parameters are incorrect, fallback to single thread to err */
-        free(params->params);
+        if (params->params)
+            free(params->params);
```
Comment (riverszhang89) [inline db/dohsql.c]: ditto

File: db/dohast.c (hunk 1 above)
Comment (dorinhogea) [reply db/dohast.c]: I didn't want to make assumptions about our code that overrides free, but sure.

**Comment by dorinhogea** [comment]
thank you

**Pull request merged**